src/cmd/services/m3coordinator/downsample/downsampler_test.go

```diff
@@ -2841,6 +2842,53 @@ func TestDownsamplerWithOverrideNamespace(t *testing.T) {
 	testDownsamplerAggregation(t, testDownsampler)
 }
 
+func TestSafeguardInProcessDownsampler(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	store := kv.NewMockStore(ctrl)
+	store.EXPECT().SetIfNotExists(gomock.Eq(matcher.NewOptions().NamespacesKey()), gomock.Any()).Return(0, nil).Times(1)
+	store.EXPECT().Set(gomock.Any(), gomock.Any()).Times(0)
+	store.EXPECT().CheckAndSet(gomock.Any(), gomock.Any(), gomock.Any()).Times(0)
+	store.EXPECT().Delete(gomock.Any()).Times(0)
+	_ = newTestDownsampler(t, testDownsamplerOptions{
+		remoteClientMock: nil,
+		kvStore:          store,
+	})
```

Comment:
Nit: slightly easier to read test code when you separate controller setup / mock setup / invocation with empty lines.